{
  "abstract": [
    {
      "text": "The error’s user-info dictionary.",
      "type": "text"
    }
  ],
  "documentVersion": 0,
  "hierarchy": {
    "paths": [
      [
        "doc://com.apple.documentation/documentation/technologies",
        "doc://com.apple.documentation/documentation/foundation",
        "doc://com.apple.documentation/documentation/foundation/url_loading_system",
        "doc://com.apple.documentation/documentation/foundation/urlerror"
      ]
    ]
  },
  "identifier": {
    "interfaceLanguage": "swift",
    "url": "doc://com.apple.documentation/documentation/foundation/urlerror/2882929-erroruserinfo"
  },
  "kind": "symbol",
  "legacy_identifier": 2882929,
  "legalNotices": {
    "copyright": "Copyright &copy; 2025 Apple Inc. All rights reserved.",
    "privacyPolicy": "https://www.apple.com/privacy/privacy-policy",
    "termsOfUse": "https://www.apple.com/legal/internet-services/terms/site.html"
  },
  "metadata": {
    "externalID": "s:10Foundation13CustomNSErrorPAAE13errorUserInfoSDySSypGvp::SYNTHESIZED::s:10Foundation8URLErrorV",
    "fragments": [
      {
        "kind": "text",
        "text": "var "
      },
      {
        "kind": "identifier",
        "text": "errorUserInfo"
      },
      {
        "kind": "text",
        "text": ": [String : Any]"
      }
    ],
    "modules": [
      {
        "name": "Foundation"
      }
    ],
    "parent": {
      "title": "URLError"
    },
    "platforms": [
      {
        "current": "18.4",
        "introducedAt": "8.0",
        "name": "iOS"
      },
      {
        "current": "18.4",
        "introducedAt": "8.0",
        "name": "iPadOS"
      },
      {
        "current": "18.4",
        "introducedAt": "13.0",
        "name": "Mac Catalyst"
      },
      {
        "current": "15.4",
        "introducedAt": "10.10",
        "name": "macOS"
      },
      {
        "current": "18.4",
        "introducedAt": "9.0",
        "name": "tvOS"
      },
      {
        "current": "2.4",
        "introducedAt": "1.0",
        "name": "visionOS"
      },
      {
        "current": "11.4",
        "introducedAt": "2.0",
        "name": "watchOS"
      },
      {
        "current": "16.3",
        "introducedAt": "9.0",
        "name": "Xcode"
      }
    ],
    "role": "symbol",
    "roleHeading": "Instance Property",
    "symbolKind": "structp",
    "title": "errorUserInfo"
  },
  "primaryContentSections": [
    {
      "declarations": [
        {
          "languages": [
            "swift"
          ],
          "platforms": [
            "iOS",
            "iPadOS",
            "Mac Catalyst",
            "macOS",
            "tvOS",
            "visionOS",
            "watchOS"
          ],
          "tokens": [
            {
              "kind": "keyword",
              "text": "var"
            },
            {
              "kind": "text",
              "text": " "
            },
            {
              "kind": "identifier",
              "text": "errorUserInfo"
            },
            {
              "kind": "text",
              "text": ": "
            },
            {
              "kind": "text",
              "text": "["
            },
            {
              "identifier": "doc://com.apple.Swift/documentation/Swift/String",
              "kind": "typeIdentifier",
              "preciseIdentifier": "s:SS",
              "text": "String"
            },
            {
              "kind": "text",
              "text": " : "
            },
            {
              "kind": "keyword",
              "text": "Any"
            },
            {
              "kind": "text",
              "text": "]"
            },
            {
              "kind": "text",
              "text": " { "
            },
            {
              "kind": "keyword",
              "text": "get"
            },
            {
              "kind": "text",
              "text": " }"
            }
          ]
        }
      ],
      "kind": "declarations"
    }
  ],
  "references": {
    "doc://com.apple.Swift/documentation/Swift/String": {
      "identifier": "doc://com.apple.Swift/documentation/Swift/String",
      "kind": "symbol",
      "role": "symbol",
      "title": "String",
      "type": "topic",
      "url": "/documentation/swift/string"
    },
    "doc://com.apple.documentation/documentation/foundation": {
      "identifier": "doc://com.apple.documentation/documentation/foundation",
      "kind": "symbol",
      "role": "collection",
      "title": "Foundation",
      "type": "topic",
      "url": "/documentation/foundation"
    },
    "doc://com.apple.documentation/documentation/foundation/url_loading_system": {
      "identifier": "doc://com.apple.documentation/documentation/foundation/url_loading_system",
      "kind": "article",
      "role": "collectionGroup",
      "title": "URL Loading System",
      "type": "topic",
      "url": "/documentation/foundation/url_loading_system"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror": {
      "abstract": [
        {
          "text": "Error codes returned by URL loading APIs.",
          "type": "text"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror",
      "kind": "symbol",
      "role": "symbol",
      "title": "URLError",
      "type": "topic",
      "url": "/documentation/foundation/urlerror"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/2292960-failureurlstring": {
      "abstract": [
        {
          "text": "The string for the URL which caused a load to fail.",
          "type": "text"
        }
      ],
      "deprecated": true,
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "failureURLString"
        },
        {
          "kind": "text",
          "text": ": String?"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/2292960-failureurlstring",
      "kind": "symbol",
      "role": "symbol",
      "title": "failureURLString",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/2292960-failureurlstring"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/2292963-failureurlpeertrust": {
      "abstract": [
        {
          "text": "The state of a failed SSL handshake.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "failureURLPeerTrust"
        },
        {
          "kind": "text",
          "text": ": SecTrust?"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/2292963-failureurlpeertrust",
      "kind": "symbol",
      "role": "symbol",
      "title": "failureURLPeerTrust",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/2292963-failureurlpeertrust"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/2293717-failingurl": {
      "abstract": [
        {
          "text": "The URL which caused a load to fail.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "failingURL"
        },
        {
          "kind": "text",
          "text": ": URL?"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/2293717-failingurl",
      "kind": "symbol",
      "role": "symbol",
      "title": "failingURL",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/2293717-failingurl"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/2882929-erroruserinfo": {
      "abstract": [
        {
          "text": "The error’s user-info dictionary.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "errorUserInfo"
        },
        {
          "kind": "text",
          "text": ": [String : Any]"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/2882929-erroruserinfo",
      "kind": "symbol",
      "role": "symbol",
      "title": "errorUserInfo",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/2882929-erroruserinfo"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/2883041-localizeddescription": {
      "abstract": [
        {
          "text": "A string containing the localized description of the error.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "localizedDescription"
        },
        {
          "kind": "text",
          "text": ": String"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/2883041-localizeddescription",
      "kind": "symbol",
      "role": "symbol",
      "title": "localizedDescription",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/2883041-localizeddescription"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/2883150-errorcode": {
      "abstract": [
        {
          "text": "The error code within the domain.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "errorCode"
        },
        {
          "kind": "text",
          "text": ": Int"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/2883150-errorcode",
      "kind": "symbol",
      "role": "symbol",
      "title": "errorCode",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/2883150-errorcode"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/3329605-backgroundtaskcancelledreason": {
      "abstract": [
        {
          "text": "The reason for canceling a background task.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "backgroundTaskCancelledReason"
        },
        {
          "kind": "text",
          "text": ": URLError.BackgroundTaskCancelledReason?"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/3329605-backgroundtaskcancelledreason",
      "kind": "symbol",
      "role": "symbol",
      "title": "backgroundTaskCancelledReason",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/3329605-backgroundtaskcancelledreason"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/3329606-downloadtaskresumedata": {
      "abstract": [
        {
          "text": "An opaque data object used to resume a failed download task.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "downloadTaskResumeData"
        },
        {
          "kind": "text",
          "text": ": Data?"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/3329606-downloadtaskresumedata",
      "kind": "symbol",
      "role": "symbol",
      "title": "downloadTaskResumeData",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/3329606-downloadtaskresumedata"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/3329607-networkunavailablereason": {
      "abstract": [
        {
          "text": "The reason the network was unavailable for a task.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "networkUnavailableReason"
        },
        {
          "kind": "text",
          "text": ": URLError.NetworkUnavailableReason?"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/3329607-networkunavailablereason",
      "kind": "symbol",
      "role": "symbol",
      "title": "networkUnavailableReason",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/3329607-networkunavailablereason"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/backgroundtaskcancelledreason": {
      "abstract": [
        {
          "text": "An enumeration of reasons used to explain the cancellation of a background task.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "enum "
        },
        {
          "kind": "identifier",
          "text": "URLError.BackgroundTaskCancelledReason"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/backgroundtaskcancelledreason",
      "kind": "symbol",
      "role": "symbol",
      "title": "URLError.BackgroundTaskCancelledReason",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/backgroundtaskcancelledreason"
    },
    "doc://com.apple.documentation/documentation/foundation/urlerror/networkunavailablereason": {
      "abstract": [
        {
          "text": "An enumeration of reasons explaining why a task couldn’t satisfy networking constraints.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "enum "
        },
        {
          "kind": "identifier",
          "text": "URLError.NetworkUnavailableReason"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/urlerror/networkunavailablereason",
      "kind": "symbol",
      "role": "symbol",
      "title": "URLError.NetworkUnavailableReason",
      "type": "topic",
      "url": "/documentation/foundation/urlerror/networkunavailablereason"
    },
    "doc://com.apple.documentation/documentation/technologies": {
      "identifier": "doc://com.apple.documentation/documentation/technologies",
      "kind": "technologies",
      "title": "Technologies",
      "type": "topic",
      "url": "/documentation/technologies"
    }
  },
  "schemaVersion": {
    "major": 0,
    "minor": 3,
    "patch": 0
  },
  "sections": [],
  "seeAlsoSections": [
    {
      "generated": true,
      "identifiers": [
        "doc://com.apple.documentation/documentation/foundation/urlerror/2883150-errorcode",
        "doc://com.apple.documentation/documentation/foundation/urlerror/2883041-localizeddescription",
        "doc://com.apple.documentation/documentation/foundation/urlerror/2293717-failingurl",
        "doc://com.apple.documentation/documentation/foundation/urlerror/2292963-failureurlpeertrust",
        "doc://com.apple.documentation/documentation/foundation/urlerror/2292960-failureurlstring",
        "doc://com.apple.documentation/documentation/foundation/urlerror/3329606-downloadtaskresumedata",
        "doc://com.apple.documentation/documentation/foundation/urlerror/3329605-backgroundtaskcancelledreason",
        "doc://com.apple.documentation/documentation/foundation/urlerror/backgroundtaskcancelledreason",
        "doc://com.apple.documentation/documentation/foundation/urlerror/3329607-networkunavailablereason",
        "doc://com.apple.documentation/documentation/foundation/urlerror/networkunavailablereason"
      ],
      "title": "Error details"
    }
  ],
  "variants": [
    {
      "paths": [
        "documentation/foundation/urlerror/2882929-erroruserinfo"
      ],
      "traits": [
        {
          "interfaceLanguage": "swift"
        }
      ]
    }
  ]
}
